src/services/maci/service.ts

```diff
@@ -505,35 +344,39 @@ export class MaciService {
       MACIFactory.abi,
       MACIFactory.linkBytecode(
         linkPoseidonLibraries(
-          poseidonT3ContractAddress,
-          poseidonT4ContractAddress,
-          poseidonT5ContractAddress,
-          poseidonT6ContractAddress,
+          POSEIDON_T3_ADDRESS,
+          POSEIDON_T4_ADDRESS,
+          POSEIDON_T5_ADDRESS,
+          POSEIDON_T6_ADDRESS,
         ),
       ),
       this.deployer,
     );
 
     const maciContract =
       await this.deployment.deployContractWithLinkedLibraries<MACI>(
-        maciContractFactory,
+        {
+          contractFactory: maciContractFactory,
+          signer: this.deployer,
+        },
         pollFactoryContractAddress,
         messageProcessorFactoryContractAddress,
         tallyFactoryContractAddress,
         gatekeeperContractAddress,
         constantInitialVoiceCreditProxyContractAddress,
-        topupCreditContractAddress,
-        stateTreeDepth,
+        STATE_TREE_DEPTH,
       );
 
-    const gatekeeperContract = await this.deployment.getContract<EASGatekeeper>(
-      {
-        name: EContracts.EASGatekeeper,
-        address: gatekeeperContractAddress,
-      },
-    );
+    const gatekeeperContract = new Contract(
+      gatekeeperContractAddress,
+      EASGatekeeperFactory.abi,
+      this.deployer,
+    ) as unknown as EASGatekeeper;
```

Comment:
Please don't rewrite it with `new Contract` and use `this.deployment.getContract` for all the contracts in this service

Reply:
You don't need to cast to unknown and you don't need to write `new Contract` everytime. Plus, it's already implemented and no need to add two different ways for creating contracts.

Reply:
<img width="625" alt="image" src="https://github.com/privacy-scaling-explorations/maci-rpgf/assets/9023842/bc0c8c08-da45-4038-8089-8fe693e6d5df">
<img width="462" alt="image" src="https://github.com/privacy-scaling-explorations/maci-rpgf/assets/9023842/6fc3cdba-6dc5-4078-843b-1efc4369a3bd">


I remembered why I used `new Contract` here, since the `getContract` method calls `hre`, which is an `undefined`. Should modify the `deployment` in maci repo first.

Reply:
If you use signer from frontend, hre won't be used

Reply:
already, but see the code below:
<img width="799" alt="image" src="https://github.com/privacy-scaling-explorations/maci-rpgf/assets/9023842/a32c69d4-b654-443c-b3e8-3d86a597bea6">

The factory is based on whether the `this.hre` exists or not, this part should be modified, else it would always return `Error(Contract ${name} not found)` here.

Reply:
So, if there is no `hre` and there is no address, abi and signer, this should be the only one case for the error.
In case if there is no `factory` just use `new BaseContract(address!, [], deployer) as unknown as T;` inside `getContract<T>`

Reply:
agree with that, would make a PR for it.